models/ItemTreeGenerator.php

```diff
@@ -0,0 +1,257 @@
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA;
+ *
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoDevTools\models;
+
+use common_Exception;
+use common_exception_Error;
+use common_ext_ExtensionException;
+use common_ext_ExtensionsManager;
+use common_report_Report;
+use core_kernel_classes_Class as RdfClass;
+use Exception;
+use oat\oatbox\extension\script\ScriptAction;
+use oat\tao\model\TaoOntology;
+use oat\taoDevTools\helper\NameGenerator;
+use oat\taoQtiItem\model\qti\ImportService;
+use RuntimeException;
+
+class ItemTreeGenerator extends ScriptAction
+{
+    private const OPTION_ITEMS_COUNT = 'items_count';
+    private const OPTION_CLASS_COUNT = 'class_count';
+    private const OPTION_NESTING_LEVEL = 'nesting_level';
+    private const OPTION_OWN_ROOT = 'own_root';
+    private const OPTION_ROOT_CLASS = 'root_class';
+
+    private $classesCount = 0;
+    private $itemsCount = 0;
+
+    protected function provideOptions(): array
+    {
+        return [
+            self::OPTION_ITEMS_COUNT   => [
+                'longPrefix'   => 'items-count',
+                'prefix'       => 'i',
+                'required'     => false,
+                'description'  => 'Number of items in class. Can be int or range. Example: 5, 1-5',
+                'defaultValue' => '2'
+            ],
+            self::OPTION_CLASS_COUNT   => [
+                'longPrefix'   => 'class-count',
+                'prefix'       => 'c',
+                'required'     => false,
+                'description'  => 'Number of classes in class. Can be int or range. Example: 5, 1-5\',',
+                'defaultValue' => '2'
+            ],
+            self::OPTION_NESTING_LEVEL => [
+                'longPrefix'   => 'nesting-level',
+                'prefix'       => 'n',
+                'required'     => false,
+                'description'  => 'Nesting level. Can be int or range. Example: 5, 1-5\',',
+                'defaultValue' => '3'
+            ],
+            self::OPTION_OWN_ROOT      => [
+                'longPrefix'   => self::OPTION_OWN_ROOT,
+                'prefix'       => 'r',
+                'required'     => false,
+                'description'  => 'All classes create under own root',
+                'defaultValue' => true
+            ],
+            self::OPTION_ROOT_CLASS    => [
+                'longPrefix'   => 'root-class',
+                'prefix'       => 'k',
+                'required'     => false,
+                'description'  => 'Root class',
+                'defaultValue' => TaoOntology::CLASS_URI_ITEM
+            ],
+        ];
+    }
+
+    protected function provideUsage(): array
+    {
+        return [
+            'prefix' => 'h',
+            'longPrefix' => 'help',
+            'description' => 'Prints a help statement'
+        ];
+    }
+
+    protected function provideDescription(): string
+    {
+        return 'Tool to generate items';
```

Comment:
Better to enhance the description to cover the class hierarchy structure generation